### ISSUE TITLE
Single-line monowidth formatting gets merged for several lines

### ISSUE THREAD

**Issue by MasterGroosha** [telegramdesktop/tdesktop#24985]
### Steps to reproduce

Possibly related to #24065. 

Start by creating a single line consisting of some monowidth blocks:
![изображение](https://user-images.githubusercontent.com/2780044/186636486-ea352fb4-ed9c-46f7-b07c-7e25c67d5893.png)
both blocks are clickable.

After you send a message, press "up" to edit it, then add some block on new line:
![изображение](https://user-images.githubusercontent.com/2780044/186637056-a693d094-c600-4f8d-ae4c-836ce4eac62b.png)

Result: 
![изображение](https://user-images.githubusercontent.com/2780044/186637270-a6055876-8ea8-4ffe-ba36-20dc451cf06e.png)

(now proceed to the next section)

### Expected behaviour

All blocks: text1, text2, text3 and text4 should be clicked separately

### Actual behaviour

Only text1 and text4 blocks are clickable, text2 and text3 blocks get merged into single multiline code block.

Also from ShowJsonBot: 
```json
"entities": [
   {
    "offset": 0,
    "length": 5,
    "type": "code"
   },
   {
    "offset": 7,
    "length": 12,
    "type": "code"
   },
   {
    "offset": 21,
    "length": 5,
    "type": "code"
   }
```

### Operating system

Manjaro

### Version of Telegram Desktop

4.1.1 (from Github releases) ; 4.9.2

### Installation source

Other (unofficial) source

**Comment by MasterGroosha** [comment]
Still reproducible with Telegram v4.6.4beta

**Comment by MasterGroosha** [comment]
Still reproducible with v4.9.2

**Comment by MasterGroosha** [comment]
Seems to be fixed in v4.15
